rates_api/app.py

```diff
@@ -12,47 +12,50 @@
 
 @app.route("/rates")
 def get_rates() -> Response:
+    """
+    Function extracts arguments from the query parameters, checks them,
+    requests data from database and forms json object
+    :param date_from: first date of the requested period
+    :param date_to: last date of the requested period
+    :param org: origin port code
+    :param dst: destination port code
+    :return: Json object included days and average prices or error info
+    """
     date_from = request.args.get("date_from", "", type=str)
     date_to = request.args.get("date_to", "", type=str)
     orig = request.args.get("origin", "", type=str)
     dest = request.args.get("destination", "", type=str)
 
-    if not dates_validator(date_from=date_from, date_to=date_to):
-        abort(404)
-
-    available_date_points = get_available_dates(date_from, date_to, orig, dest)
-    if not available_date_points:
-        abort(404)
-    else:
-        date_from_f, date_to_f = available_date_points
-        daily_rates = get_daily_rates(orig, dest)
-        if not daily_rates:
-            abort(404)
-        dates_range = get_dates(str(date_from_f), str(date_to_f))
-        rates = []
-        flag = 0
-        for day in dates_range:
-            for daily_rate in daily_rates:
-                if day == str(daily_rate["day"]):
-                    rate = {
-                        "day": str(daily_rate["day"]),
-                        "average_price": int(daily_rate["rate"]),
-                    }
-                    rates.append(rate)
-                    flag = 1
-                    break
-            if not flag:
-                rate = {"day": day, "average_price": None}
-                rates.append(rate)
-            flag = 0
-
-        return jsonify(rates)
-
-
-@app.errorhandler(404)
+    rates = []
+    date_validation_result = dates_validator(date_from, date_to)
+
+    # If at least one of dates is incorrect or date from > date_to:
+    if not date_validation_result:
+        abort(422)
+
+    # Valid dates converted into date format:
+    date_from_f, date_to_f = date_validation_result
+
+    # List of dicts included days and average prices from database:
+    daily_rates = get_daily_rates(date_from_f, date_to_f, orig, dest)
+
+    for daily_rate in daily_rates:
+        day = str(daily_rate["day"])
+        avg_price = daily_rate["avg_price"]
+        rate = {
+            "day": day,
+            "average_price": avg_price,
+        }
+        rates.append(rate)
```

Comment:
wdyt, would it make it clearer if this operation was performed as list comprehension?